src/test/java/cms/gov/madie/measure/resources/MeasureTransferControllerTest.java

```diff
@@ -416,4 +437,34 @@ public void verifyIfOrganizationNamesAreUpdated() {
     assertEquals(
         "Innovaccer Url", persistedMeasure.getMeasureMetaData().getDevelopers().get(2).getUrl());
   }
+
+  @Test
+  public void testReorderGroupPopulations() {
+    Group copiedGroup = Group.builder().populations(groups.get(0).getPopulations()).build();
```

Comment:
Why are we creating a new copiedGroup with the existing values ?

Reply:
@RohitKandimalla Just to show the difference in ordering.

Reply:
We could use the input itself

Reply:
@RohitKandimalla The input itself will be reordered, I wanted to show the comparison before and after like this: assertEquals(
        copiedGroup.getPopulations().**get(2)**.getId(),
        reorderedGroups.get(0).getPopulations().**get(5)**.getId());

Reply:
> Wouldn't it be safer to keep the original
The general consensus is, yes, defensive copying of objects is preferred to avoid unexpected side effects should the method be used elsewhere or the class structure change. 

**An even better option is to design immutable classes.**  Thereby removing the ability to mutate objects as they're passed around.

The counter offer is performance and convenience. If the defensive copy is of an immense data structure or collection, it might problematic to copy on every method invocation. Or if the use case is well-known, the objects at play are well-behaved, and the method signature's name conveys the expectation of mutation, then it's not terrible.

Since this is a unit test and JUnit is resetting the `Groups` object between tests, it isn't a major issue.